typescript/agentkit/src/wallet-providers/privyEvmEmbeddedWalletProvider.ts

```diff
@@ -0,0 +1,537 @@
+   * @param params.address - The address of the contract
+   * @param params.abi - The ABI of the contract
+   * @param params.functionName - The name of the function to call
+   * @param params.args - The arguments to pass to the function
+   * @returns A promise that resolves to the contract function's return value
+   */
+  async readContract<
+    const abi extends Abi | readonly unknown[],
+    functionName extends ContractFunctionName<abi, "pure" | "view">,
+    const args extends ContractFunctionArgs<abi, "pure" | "view", functionName>,
+  >(
+    params: ReadContractParameters<abi, functionName, args>,
+  ): Promise<ReadContractReturnType<abi, functionName, args>> {
+    const body = {
+      address: this.#address,
+      chain_type: "ethereum",
+      chain_id: this.mapNetworkToChainId(this.#network.networkId!),
+      contract: {
+        address: params.address,
+        abi: params.abi,
+        functionName: params.functionName,
+        args: params.args,
+      },
+    };
+
+    const response = await this.executePrivyRequest<{
+      result: ReadContractReturnType<abi, functionName, args>;
+    }>({
+      method: "eth_call",
+      params: body,
+    });
+    return response.result;
+  }
+
+  /**
+   * Transfer the native asset of the network.
+   *
+   * @param to - The destination address.
+   * @param value - The amount to transfer in Wei.
+   * @returns The transaction hash.
+   */
+  async nativeTransfer(to: string, value: string): Promise<`0x${string}`> {
+    const valueInWei = parseEther(value);
+    const valueHex = `0x${valueInWei.toString(16)}`;
+
+    const body = {
+      address: this.#address,
+      chain_type: "ethereum",
+      method: "eth_sendTransaction",
+      caip2: `eip155:${this.mapNetworkToChainId(this.#network.networkId!)}`,
+      params: {
+        transaction: {
+          to,
+          value: valueHex,
+        },
+      },
+    };
+
+    try {
+      const response = await this.executePrivyRequest<PrivyResponse<{ hash: `0x${string}` }>>(body);
+
+      const receipt = await this.waitForTransactionReceipt(response.data.hash);
+
+      if (!receipt) {
+        throw new Error("Transaction failed");
+      }
+
+      return receipt.transactionHash;
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Native transfer failed: ${error.message}`);
+      }
+      throw new Error("Native transfer failed");
+    }
+  }
+
+  /**
+   * Exports the wallet information.
+   *
+   * @returns The wallet data
+   */
+  exportWallet(): PrivyWalletExport {
+    return {
+      walletId: this.#walletId,
+      authorizationPrivateKey: this.#authKey,
+      networkId: this.#network.networkId!,
+      chainId: this.#network.chainId,
+    };
+  }
+
+  /**
+   * Map network IDs to their corresponding chain IDs
+   *
+   * @param networkId - The network ID to map
+   * @returns The corresponding chain ID
+   */
+  private mapNetworkToChainId(networkId: string): string {
```

Comment:
Can be removed after removing the references and leveraging `NETWORK_ID_TO_CHAIN_ID` in the constructor